src/Common/src/System/Security/Cryptography/DerSequenceReader.cs

```diff
@@ -110,6 +110,67 @@ internal byte[] ReadNextEncodedValue()
             return encodedValue;
         }
 
+        internal byte[] ReadRemainingData()
+        {
+            int remainingBytes = _end - _position;
+            byte[] remainingData = new byte[remainingBytes];
+            Buffer.BlockCopy(_data, _position, remainingData, 0, remainingBytes);
+            return remainingData;
+        }
+
+        private static byte[][] SplitValue(byte[] payload, ref int offset)
+        {
+            int lengthLength;
+            int contentLength = ScanContentLength(payload, offset + 1, out lengthLength);
+
+            byte[] encodedLength = new byte[lengthLength];
```

Comment:
There seems to be a lot of array allocation/copying going on in the codebase, with lots of byte[][] getting passed around.  This may be necessary given the current structure... at some point it'd be good to revisit all of this to see if there's a structure that would afford less.